.github/workflows/release.yaml

```
@@ -10,13 +10,42 @@ on:
+  # For pushes to main the publishing flow is two-phase (per-arch matrix then
+  # sequential manifest assembly for both components). Cancelling mid-run can
+  # leave satellite:latest and ground-control:latest pointing at different
+  # commits with nothing to repair the inconsistency, so we let main-branch
+  # runs queue instead of cancelling each other.
+  # For pull requests we keep cancel-in-progress: true for fast feedback.
+  group: ${{ github.ref == 'refs/heads/main' && 'publish-main' || format('ci-{0}', github.ref_name) }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
 jobs:
+  setup:
+    runs-on: container-registry
```

actionlint (1.7.12)

[error] label "container-registry" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)